File: pocket-core/src/main/java/com/dididi/pocket/core/delegates/bottom/BottomTabBean.java
package com.dididi.pocket.core.delegates.bottom;

import androidx.annotation.Nullable;

/**
 * Created by dididi
 * on 24/07/2018 .
 */

public final class BottomTabBean {
    //存储bottomBar的tab
    /**
     * CharSequence是String,StringBuilder,StringBuffer的接口
     */
    private final CharSequence ICON;
    private final CharSequence TITLE;

    public BottomTabBean(CharSequence icon, CharSequence title) {
        this.ICON = icon;
        this.TITLE = title;
    }

    public CharSequence getIcon() {
        return ICON;
    }

    public CharSequence getTitle() {
        return TITLE;
    }

    @Override
    public boolean equals(@Nullable Object obj) {
        if (obj instanceof BottomTabBean) {
            BottomTabBean item = (BottomTabBean) obj;
            return this.ICON.equals(item.getIcon()) && this.TITLE.equals(item.getTitle());
        }
        return false;
    }

    @Override
    public int hashCode() {
        return this.ICON.hashCode() + (this.TITLE.hashCode()) * 3;
    }
}

File: pocket-core/src/main/java/com/dididi/pocket/core/delegates/bottom/BottomTabBean.kt
package com.dididi.pocket.core.delegates.bottom

/**
 * @author dididi
 * on 24/07/2018 .
 * @describe 存储bottomBar的tab
 */

class BottomTabBean(val icon: CharSequence, val title: CharSequence) {

    override fun hashCode(): Int {
        return icon.hashCode() * 2 + title.hashCode() * 3
    }

    override fun equals(other: Any?): Boolean {
        if (other is BottomTabBean){
            return icon == other.icon && title == other.title
        }
        return false
    }
}
